src/openforms/submissions/logic/actions.py

```diff
@@ -308,6 +309,57 @@ def get_action_log_data(
         }
 
 
+class DMNConfig(TypedDict):
+    plugin_id: str
+    input_mapping: dict[str, str]
+    output_mapping: dict[str, str]
+    definition_id: str
+    definition_version: str
+
+
+@dataclass
+class EvaluateDMNAction(ActionOperation):
+    input_mapping: dict[str, str]
+    output_mapping: dict[str, str]
+    definition_id: str
+    plugin_id: str
+    definition_version: str = ""
+
+    @classmethod
+    def from_action(cls, action: ActionDict) -> "ActionOperation":
+        dmn_config: DMNConfig = action["action"]["value"]
+
+        return cls(**dmn_config)
+
+    def eval(
+        self,
+        context: DataMapping,
+        log: Callable[[JSONValue], None],
+        submission: Submission,
+    ) -> DataMapping | None:
+        # Mapping from form variables to DMN inputs
+        dmn_inputs = {
+            key_dmn: glom(context, key_form)
+            for key_form, key_dmn in self.input_mapping.items()
+        }
+
+        # Perform DMN call
+        dmn_outputs = evaluate_dmn(
+            definition_id=self.definition_id,
+            version=self.definition_version,
+            input_values=dmn_inputs,
+            plugin_id=self.plugin_id,
+        )
+
+        log({"dmn_inputs": dmn_inputs, "dmn_outputs": dmn_outputs})
+
+        # Map DMN output to form variables
+        return {
+            key_form: dmn_outputs[key_dmn]
+            for key_form, key_dmn in self.output_mapping.items()
+        }
```

Comment:
does this return value work correctly with deep keys like `foo.bar`?

Reply:
The second test added in src/openforms/submissions/tests/form_logic/test_modify_variables.py (line 503) tests the nested variable!